core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/ComponentParser.java

```diff
@@ -0,0 +1,307 @@
+                    return new ArrayList<>(content.keySet());
+                }
+            }
+        }
+        return new ArrayList<>();
+    }
+
+    public List<Map<String, String>> generateNames(List<List<String>> lists, List<Map<String, String>> result, int depth, Map<String, String> current) {
+        if (depth == lists.size()) {
+            result.add(new HashMap<>(current));
+            return result;
+        }
+        if (lists.get(depth).isEmpty()) {
+            generateNames(lists, result, depth + 1,  createName(null, current, depth));
+        }
+        for (int i = 0; i < lists.get(depth).size(); i++) {
+            String value = lists.get(depth).get(i);
+            generateNames(lists, result, depth + 1,  createName(value, current, depth));
+        }
+
+        return result;
+    }
+
+    public Map<String, String> createName(String value,  Map<String, String> current, int depth) {
+        switch (depth) {
+            case 0:
+                current.put("security", value);
+                break;
+            case 1:
+                current.put(REQUEST, value);
+                break;
+            case 2:
+                current.put(RESPONSE, value);
+                break;
+            default:
+                break;
+        }
+        return current;
+    }
+
+
+    public  Component createComponent(Map<String, String> partNames, Operation operation,PathItem.HttpMethod operationName, String pathName, OpenAPI openAPI) {
+        long version = Long.parseLong(openAPI.getInfo().getVersion());
+        String connectionName = openAPI.getInfo().getTitle();
+        String componentName = buildName(operation.getOperationId(), partNames);
+        String componentType = "http.request";
+        String componentDescription = operation.getDescription();
+        ComponentVersion componentVersion = new ComponentVersion(new ComponentVersionKey(componentName, version),componentDescription);
+
+        List<ComponentParameter> info = getInfo(componentName,pathName, operationName, version, connectionName);
+        List<ComponentParameter> queryParams = getQueryParams(componentName, operation.getParameters(), version);
+        List<ComponentParameter> headers = getHeaders(componentName, partNames, operation, openAPI);
+        List<ComponentParameter> params = Stream.of(info, queryParams, headers).flatMap(Collection::stream).collect(Collectors.toList());
+        return new Component(new ComponentKey(componentName, version), componentType, componentName, componentDescription, componentVersion, params, new ArrayList<>());
+
+    }
+
+    public  List<ComponentParameter> getInfo(String componentName, String pathName, PathItem.HttpMethod operationName, Long version, String connectionName) {
+        List<ComponentParameter> componentParameters = new ArrayList<>();
+        componentParameters.add(new ComponentParameter(new ComponentParameterKey(componentName, version, "endpoint"), pathName));
```

Comment:
path name of format /test/{id} is not transformed to /test/#id#. 
Also, write a unit test for this